### PR TITLE
conmon: detect cgroup2 and skip OOM handling

### PR DESCRIPTION
**Pull request by giuseppe**
add detection for a cgroup v2 unified mode.  Skip setting up OOM for
now.  It is not proper support for will temporarily unlock Podman.

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: conmon/conmon.c [hunk 1]
```diff
@@ -878,9 +880,17 @@ static int setup_terminal_control_fifo()
 static void setup_oom_handling(int container_pid)
 {
 	/* Setup OOM notification for container process */
-	_cleanup_free_ char *memory_cgroup_path = process_cgroup_subsystem_path(container_pid, "memory");
+	_cleanup_free_ char *memory_cgroup_path = NULL;
 	_cleanup_close_ int cfd = -1;
+	struct statfs sfs;
 	int ofd = -1; /* Not closed */
+
+        if (statfs ("/sys/fs/cgroup", &sfs) == 0 && sfs.f_type == CGROUP2_SUPER_MAGIC) {
```
Comment (haircommander) [inline conmon/conmon.c]: will the statfs call complain if conmon is called as non-root?

**Comment by haircommander** [comment]
LGTM

File: conmon/conmon.c (hunk 1 above)
Comment (giuseppe) [reply conmon/conmon.c]: no, it works well.  I've tested it locally only as non root :-)

File: conmon/conmon.c [hunk 2]
```diff
@@ -35,6 +37,10 @@
 #include "cmsg.h"
 #include "config.h"
 
+#ifndef CGROUP2_SUPER_MAGIC
+# define CGROUP2_SUPER_MAGIC 0x63677270
```
Comment (mheon) [inline conmon/conmon.c]: Space between `#` and `define`?

Comment (giuseppe) [reply conmon/conmon.c]: I usually use it to indent C preprocessor macros (as done by https://www.gnu.org/software/cppi/)

**Comment by rhatdan** [comment]
LGTM

**Comment by mrunalp** [comment]
@giuseppe clang format failed on the file - https://travis-ci.com/cri-o/cri-o/jobs/199234610

File: conmon/conmon.c (hunk 2 above)
Comment (giuseppe) [reply conmon/conmon.c]: well clang doesn't like it, so I had to drop it :-)

**Pull request merged**